torchmetrics/metric.py

```diff
@@ -240,6 +237,33 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return self._forward_cache
 
+    def _reduce_state(self, state_to_reduce: Dict[str, Any]) -> None:
+        for attr in self._defaults.keys():
```

Comment:
Can we actually take care of the ordering here? So that it's always `state_to_reduce + current_state` and never the other way around? This doesn't matter for tensors or floats or something like that but it is essential for list-states as `[1] + [2] != [2] + [1]`. Personally I have sliding window states, where whenever the window size limit is hit, the first element is deleted/overwritten. Therefore ordering matters. Same with concatenation of tensors.